### PR TITLE
add config option to raise error when outbox entry record is not found

### PR DESCRIPTION
**Pull request by addicted2sounds**
it should help to avoid raising errors for deleted records

### CODE REVIEW

File: lib/rails_transactional_outbox/configuration.rb [hunk 1]
```diff
@@ -35,6 +36,10 @@ def outbox_entries_processor
       @outbox_entries_processor ||= RailsTransactionalOutbox::OutboxEntriesProcessors::NonOrderedProcessor.new
     end
 
+    def raise_not_found_model_error
+      @raise_not_found_model_error.nil? ? true : @raise_not_found_model_error
```
Comment (Azdaroth) [inline lib/rails_transactional_outbox/configuration.rb]: this is a bit confusing...

How about doing it that way:

```
def raise_not_found_model_error
  return @raise_not_found_model_error if defined?(@raise_not_found_model_error)

  true
end

alias_method :raise_not_found_model_error?, :raise_not_found_model_error
```

File: lib/rails_transactional_outbox/record_processors/active_record_processor.rb [hunk 2]
```diff
@@ -15,7 +15,12 @@ def applies?(record)
       end
 
       def call(record)
-        model = record.infer_model or raise CouldNotFindModelError.new(record)
+        model = record.infer_model
+        if model.nil?
+          return unless RailsTransactionalOutbox.configuration.raise_not_found_model_error
```
Comment (Azdaroth) [inline lib/rails_transactional_outbox/record_processors/active_record_processor.rb]: using `if` is more readable, how about doing it the other way round like:
```
raise CouldNotFindModelError.new(record) if raise_not_found_model_error?

return
```

**Review by Azdaroth — COMMENTED** [review]
Can you also please update the Changeling and Readme?

File: lib/rails_transactional_outbox/record_processors/active_record_processor.rb (hunk 2 above)
Comment (addicted2sounds) [reply lib/rails_transactional_outbox/record_processors/active_record_processor.rb]: raise with condition happens to be a long line :) `unless` is pretty readable for simple conditions like that. Anyway adjusted

**Comment by addicted2sounds** [comment]
@Azdaroth bumped version as well. Lmk if that should be done in separate PR

**Review by Azdaroth — APPROVED** [review]
💪 🐉 Opa!

**Pull request merged**